pennylane/qchem/observable_hf.py

```diff
@@ -116,6 +125,18 @@ def qubit_observable(o_ferm, cutoff=1.0e-12):
                 ops = ops + op[1]
                 coeffs = qml.math.concatenate([coeffs, qml.math.array(op[0]) * o_ferm[0][n]])
 
+    if active_new_opmath():
+        arithmetic_ops = [qml.prod(*op.obs) if isinstance(op, Tensor) else op for op in ops]
+
+        terms = tuple(qml.s_prod(coeff, op) for coeff, op in zip(coeffs, arithmetic_ops) if abs(coeff) >= cutoff)
+        num_terms = len(terms)
+
+        if num_terms == 0:
+            return qml.s_prod(0, qml.Identity(ops[0].wires[0]))  # use any op and any wire to represent the null op
+        if num_terms == 1:
+            return terms[0]  # just return the op
+        return qml.sum(*terms)
```

Comment:
Could potentially use `qml.dot` in this circumstance.

Reply:
😮! When did we add this! Very useful, I updated the code to use this cut the number of lines in half!